node/narwhal/src/worker.rs

```diff
@@ -365,40 +365,32 @@ impl<N: Network> Worker<N> {
 mod prop_tests {
     use super::*;
 
-    use crate::{helpers::storage::prop_tests::StorageInput, prop_tests::GatewayInput};
-    use test_strategy::{proptest, Arbitrary};
+    use test_strategy::proptest;
 
     type CurrentNetwork = snarkvm::prelude::Testnet3;
 
-    #[derive(Arbitrary, Debug, Clone)]
-    pub struct WorkerInput {
-        pub id: u8,
-        #[filter(GatewayInput::is_valid)]
-        pub gateway: GatewayInput,
-        pub storage: StorageInput,
-    }
-
-    impl WorkerInput {
-        fn to_worker(&self) -> Result<Worker<CurrentNetwork>> {
-            Worker::new(self.id, self.gateway.to_gateway(), self.storage.to_storage(), Default::default())
-        }
-
-        fn is_valid(&self) -> bool {
-            self.id < MAX_WORKERS
-        }
+    #[proptest]
+    fn worker_initialization(
+        #[strategy(0..MAX_WORKERS)] id: u8,
+        gateway: Gateway<CurrentNetwork>,
+        storage: Storage<CurrentNetwork>,
+    ) {
+        let worker = Worker::new(id, gateway, storage, Default::default());
+        assert!(worker.is_ok());
+        let worker = worker.unwrap();
+        assert_eq!(worker.id(), id);
     }
 
     #[proptest]
-    fn worker_initialization(input: WorkerInput) {
-        match input.to_worker() {
-            Ok(worker) => {
-                assert!(input.is_valid());
-                assert_eq!(worker.id(), input.id);
-            }
-            Err(e) => {
-                assert!(!input.is_valid());
-                assert_eq!(e.to_string().as_str(), format!("Invalid worker ID '{}'", input.id));
-            }
+    fn invalid_worker_id(
+        #[strategy(MAX_WORKERS..)] id: u8,
+        gateway: Gateway<CurrentNetwork>,
+        storage: Storage<CurrentNetwork>,
+    ) {
+        let worker = Worker::new(id, gateway, storage, Default::default());
+        assert!(worker.is_err());
+        if let Err(error) = worker {
+            assert_eq!(error.to_string().as_str(), format!("Invalid worker ID '{}'", id));
```

Comment:
```suggestion
            assert_eq!(error.to_string(), format!("Invalid worker ID '{}'", id));
```
this should be redundant?

Reply:
Good catch, fixing it with https://github.com/AleoHQ/snarkOS/pull/2537#discussion_r1262901582